content/documentation/pages/1-installation/1-local/1-docker.md

```diff
@@ -6,93 +6,154 @@ description: 'Installation using Docker Compose'
+| 7577        | 7577            | The port that the Skipper server listens on. You can use it to reach the Skipper REST api at http://localhost:7577/api                                                                                                                                                  |
+| 20000-20105 | 20000-20105     | Skipper and Local Deployer are configured to use this port range for all deployed stream applications. That means you can reach the application's actuator endpoints from your host machine. The `server.port` deployment property can be used to override those ports. |
+
+[[tip]]
+| You can leverage the exposed application ports (e.g. `20000-20105`) in you stream applications to expose certain ports to the host machine. For example, the `http --server.port=20015 | log` stream definition would permit you use `curl` and POST HTTP messages to the `http` source directly from your host machine on the `20015` port.
 
 ## Stopping Spring Cloud Data Flow
 
-When you want to shut down Spring Cloud Data Flow, you can use the `docker-compose down` command. To do so:
+1. Press `Ctrl+C` to shut down the docker compose process.
 
-1. Open a new terminal window.
+1. Run the following command to clean the used Docker containers:
 
-1. Change directory to the directory in which you started (where the
-   `docker-compose.yml` file is located).
+```bash
+docker-compose down
+```
+
+If errors occur due to old or hanging containers, clean all containers:
+
+<!--TABS-->
 
-1. Run the following command:
+<!--Linux / OSX / Windows (PowerShell)-->
 
 ```bash
-DATAFLOW_VERSION=%dataflow-version% SKIPPER_VERSION=%skipper-version% docker-compose down
+docker stop $(docker ps -a -q)
+docker rm $(docker ps -a -q)
 ```
 
-[[note]]
-| You need to specify the `DATAFLOW_VERSION` and the `SKIPPER_VERSION` environment variables if you are running the command in a separate terminal window from which you started Docker Compose. The `export` commands you used earlier set the variables for only that terminal window, so those values are not found in the new terminal window.
+<!--Windows (CommandPrompt)-->
 
-If all else fails, you can shut down Docker Compose by pressing `Ctrl+C`.
+```basic
+FOR /f "tokens=*" %i IN ('docker ps -aq') DO docker rm %i -f
+```
+
+<!--END_TABS-->
 
 ## Shell
 
-For convenience and as an alternative to using the Spring Cloud Data Flow Dashboard, you can use the Spring Cloud Data Flow Shell as an alternative to the UI.
-The shell supports tab completion for commands and stream and task application configuration properties.
+For convenience and as an alternative to the Spring Cloud Data Flow Dashboard, you can use the [Spring Cloud Data Flow Shell](%currentPath%/concepts/tooling/#shell).
+Later supports tab completion for commands and application configuration properties.
```

Comment:
`Later` is incorrect here.  You might mean `The latter`.  Or revert to `The shell` is the most clear.